ui/desktop/tests/acceptance/authentication-test.js

```diff
@@ -235,4 +249,108 @@ module('Acceptance | authentication', function (hooks) {
     assert.dom('[data-test-no-auth-methods]').includesText('No Auth Methods');
     assert.dom('.change-origin').exists();
   });
+
+  test('signing out with running sessions renders signout modal', async function (assert) {
+    this.ipcStub.withArgs('hasRunningSessions').returns(true);
+
+    await visit(urls.authenticate.methods.global);
+
+    await fillIn('[name="identification"]', 'test');
+    await fillIn('[name="password"]', 'test');
+    await click('[type="submit"]');
+
+    assert.ok(currentSession().isAuthenticated);
+
+    await click(
+      '.rose-header-utilities .header-dropdown-button-override button',
+    );
+
+    await click(SIGNOUT_BTN_SELECTOR);
+
+    assert.dom(CLOSE_SESSIONS_MODAL_SELECTOR).isVisible();
+    assert
+      .dom(CLOSE_SESSIONS_MODAL_SELECTOR)
+      .includesText('Sign out of Boundary?');
+
+    await click(CANCEL_BUTTON);
+
+    assert.dom(CLOSE_SESSIONS_MODAL_SELECTOR).isNotVisible();
+    assert.ok(currentSession().isAuthenticated);
+  });
+
+  test('confirming signout via modal stops sessions and logs out user', async function (assert) {
+    const stopAllSessions = this.ipcStub.withArgs('stopAll');
+    this.ipcStub.withArgs('hasRunningSessions').returns(true);
+
+    await visit(urls.authenticate.methods.global);
+
+    await fillIn('[name="identification"]', 'test');
+    await fillIn('[name="password"]', 'test');
+    await click('[type="submit"]');
+
+    assert.ok(currentSession().isAuthenticated);
+
+    await click(
+      '.rose-header-utilities .header-dropdown-button-override button',
+    );
+
+    await click(SIGNOUT_BTN_SELECTOR);
+
+    assert.dom(CLOSE_SESSIONS_MODAL_SELECTOR).isVisible();
+    assert
+      .dom(CLOSE_SESSIONS_MODAL_SELECTOR)
+      .includesText('Sign out of Boundary?');
+
+    await click(CONFIRM_BUTTON);
+
+    assert.dom(CLOSE_SESSIONS_MODAL_SELECTOR).isNotVisible();
+    assert.ok(stopAllSessions.calledOnce);
+    assert.notOk(currentSession().isAuthenticated);
+  });
+
+  test('attempting to quit app when signout modal is present triggers the close sessions modal', async function (assert) {
+    window.electron = {
+      onAppQuit: (e) => {
```

Comment:
```suggestion
      onAppQuit: (callback) => {
```
I think the `e` usually signifies an `event` or `error`, I think renaming this argument to `callback` would be clearer